src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable code-import-patterns */
+/* eslint-disable header/header */
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import { isWeb } from 'vs/base/common/platform';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { RemoteTrackMessage } from '@gitpod/gitpod-protocol/lib/analytics';
+import { ITelemetryAppender } from 'vs/platform/telemetry/common/telemetryUtils';
+
+export class GitpodInsightsAppender implements ITelemetryAppender {
+	private baseProperties: { appName: string; uiKind: 'web' | 'desktop'; version: string };
+	constructor(
+		@IProductService private readonly productService: IProductService
+	) {
+
+		this.baseProperties = {
+			appName: this.productService.nameLong,
+			uiKind: isWeb ? 'web' : 'desktop',
```

Comment:
This will always be `web`

```suggestion
			uiKind: 'web',
```